src/trackinfoobject.cpp

```diff
@@ -43,40 +43,37 @@
 #include "library/coverartutils.h"
 #include "util/assert.h"
 
-TrackInfoObject::TrackInfoObject(const QString& file,
-                                 SecurityTokenPointer pToken,
-                                 bool parseHeader, bool parseCoverArt)
-        : m_fileInfo(file),
-          m_pSecurityToken(pToken.isNull() ? Sandbox::openSecurityToken(
-                  m_fileInfo, true) : pToken),
-          m_qMutex(QMutex::Recursive),
-          m_analyserProgress(-1) {
-    initialize(parseHeader, parseCoverArt);
-}
-
 TrackInfoObject::TrackInfoObject(const QFileInfo& fileInfo,
                                  SecurityTokenPointer pToken,
                                  bool parseHeader, bool parseCoverArt)
         : m_fileInfo(fileInfo),
           m_pSecurityToken(pToken.isNull() ? Sandbox::openSecurityToken(
                   m_fileInfo, true) : pToken),
-          m_qMutex(QMutex::Recursive),
-          m_analyserProgress(-1) {
+          m_bDeleteOnReferenceExpiration(false),
+          m_qMutex(QMutex::Recursive) {
     initialize(parseHeader, parseCoverArt);
 }
 
+namespace {
+    QFileInfo parseFileInfo(const QDomNode& nodeHeader) {
+        QString filename = XmlParse::selectNodeQString(nodeHeader, "Filename");
+        QString location = QDir(XmlParse::selectNodeQString(nodeHeader, "Filepath")).filePath(filename);
+        return QFileInfo(location);
+    }
```

Comment:
Then let's rid of this legacy code entirely!

No one is able to test it and you could use one of the versions up to 1.12 
if you need to migrate a library from 1.7.

On 09/20/2015 06:11 PM, Daniel Schürmann wrote:
>
> In src/trackinfoobject.cpp 
> <https://github.com/mixxxdj/mixxx/pull/718#discussion_r39931406>:
>
> >      initialize(parseHeader, parseCoverArt);
> >  }
> >
> > +namespace {
> > +    QFileInfo parseFileInfo(const QDomNode& nodeHeader) {
> > +        QString filename = XmlParse::selectNodeQString(nodeHeader, "Filename");
> > +        QString location = QDir(XmlParse::selectNodeQString(nodeHeader, "Filepath")).filePath(filename);
> > +        return QFileInfo(location);
> > +    }
>
> We are probably not able to test a 1.7 upgrade.
> IMHO we should issue a warning, that a 1.7 library is too old and that the 
> user should install Mixxx 1.12 fist, which still includes the upgrade code.
>
> Any thought?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/718/files#r39931406>.
>